android/src/main/java/cl/json/social/SingleShareIntent.java

```diff
@@ -58,13 +58,20 @@ public void open(ReadableMap options) throws ActivityNotFoundException {
         super.open(options);
     }
 
-    protected void openIntentChooser() throws ActivityNotFoundException {
+    protected void openIntentChooser(ReadableMap options) throws ActivityNotFoundException {
         if (this.options.hasKey("forceDialog") && this.options.getBoolean("forceDialog")) {
             Activity activity = this.reactContext.getCurrentActivity();
             if (activity == null) {
                 TargetChosenReceiver.sendCallback(false, "Something went wrong");
                 return;
             }
+            String socialType = options.getString("social");
```

Comment:
seems like you could avoid touching all the other share objects if you checked for null options first here, then kept the empty-argument openIntentChooser() and just chained it to this one passing null e.g.

```java
protected void openIntentChooser() throws ActivityNotFoundException {
    this.openIntentChooser(null);
}
```

That's a minor point though

Reply:
Good idea! I'll add the check. Thanks!